stream.go

```diff
@@ -226,6 +234,19 @@ func (cs *clientStream) CloseSend() (err error) {
 	return
 }
 
+func (cs *clientStream) finish(err error) {
```

Comment:
Yes, this would break because of that. I couldn't figure out how this code should clearly delimit when a clientStream is "done", since both SendMsg and RecvMsg appear to indicate the termination of the stream when things fail. 

Does `clientStream` need a mutex to mediate this? What's supposed to happen if concurrent SendMsg/RecvMsg calls are running?